tool/readme_scripts/readme_scripts_runner.dart

```diff
@@ -18,144 +18,193 @@
 import 'dart:io';
 
 const checkmark = '\u2713';
+const crossmark = '\u2716';
 const lReadme = 'readme.md';
 const lMetadata = 'readme.metadata.json';
+const samplesDir = 'lib/samples';
+final failedChecks = <FailedCheck>[];
 
 /// Pass in the directory to a sample to run the checks
-/// dart run readme_scripts_runner.dart "path/to/samples/display_map"
+/// dart run tool/readme_scripts/readme_scripts_runner.dart display_map
+/// Or pass -all to run the checks on all samples
+/// dart run tool/readme_scripts/readme_scripts_runner.dart -all
 /// If creating a new metadata file from an existing readme, pass in the category as the second arg:
-/// dart run readme_scripts_runner.dart "path/to/samples/display_map" "Maps"
+/// dart run tool/readme_scripts/readme_scripts_runner.dart display_map "Maps"
 /// The list of valid categories can be found at common-samples/designs/categories.md
 void main(List<String> args) {
-  var filePaths = <String>[];
-  String? dirPath;
+  final sampleDirectories = <Directory>[];
+  String? currentDirPath;
   String? readmeFile;
   String? metadataFile;
   String? category;
+
   print('** Checking for sample files **');
 
-  // Get all the files in the provided path.
   if (args.isNotEmpty) {
-    dirPath = args[0];
+    Directory? directory;
+    if (args[0] == '-all') {
+      // If -all flag passed, check all samples.
+      directory = Directory(samplesDir);
+      checkIfDirExists(directory);
+      sampleDirectories.addAll(
+        directory.listSync().whereType<Directory>().map((f) => f).toList(),
```

Comment:
Is the `.map((f) => f)` part necessary? I don't think it does anything.

Reply:
Oh nice no it doesn't, whereType returns them 👍